### PR TITLE
Multi gpu

### PR DESCRIPTION
**Pull request by ZeroAct**
# Add
- add multi gpu support for `UltralyticsHub`
  - just give comma sperated device ids

```python
train(
        dataset_path=export_dir,
        device="0,1",
        epochs=100
    )
```

# TODO
- I've added custom github action runner using our dev pc (4090 x2). But its caching is extremely slow... I'm considering [this](https://buildjet.com/for-github-actions/blog/launch-buildjet-cache).

### CODE REVIEW

File: waffle_hub/hub/adapter/ultralytics/ultralytics_hub.py [hunk 1]
```diff
@@ -260,23 +262,30 @@ def on_train_start(self, cfg: TrainConfig):
 
     def training(self, cfg: TrainConfig, callback: TrainCallback):
 
-        model = YOLO(cfg.pretrained_model, task=self.backend_task_name)
+        code = f"""if __name__ == "__main__":
```
Comment (dustjr1565) [inline waffle_hub/hub/adapter/ultralytics/ultralytics_hub.py:265]: Wouldn't it be better to create a separate function? It looks strange as it is.
like as there include " for a string type,

File: .github/workflows/ddp.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    branches: ["*"]
+
+jobs:
+  Tests:
+    timeout-minutes: 60
+    runs-on: self-hosted
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.9']
```
[the remark this answers is not in the document]
Comment (ZeroAct) [reply .github/workflows/ddp.yaml:14]: We don't have windows dev server...😂

File: waffle_hub/utils/evaluate.py [hunk 3]
```diff
@@ -1,10 +1,11 @@
-from typing import Union
 from functools import reduce
 from operator import eq
+from typing import Union
 
 import torch
 from torchmetrics.classification import Accuracy
 from torchmetrics.detection import mean_ap
+from torchmetrics.text import CharErrorRate
```
[the remark this answers is not in the document]
Comment (ZeroAct) [reply waffle_hub/utils/evaluate.py:8]: It has occured when i merge dev branch.
My mistake.

File: waffle_hub/hub/adapter/ultralytics/ultralytics_hub.py (hunk 1 above)
Comment (ZeroAct) [reply waffle_hub/hub/adapter/ultralytics/ultralytics_hub.py:265]: I cannot understand what you saying.

**Pull request merged**